packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx

```diff
@@ -76,9 +87,28 @@ const MessageBubble = (props: ChatMessageComponentAsMessageBubbleProps): JSX.Ele
     onRenderAvatar,
     showMessageStatus,
     messageStatus,
-    fileDownloadHandler
+    fileDownloadHandler,
+    /* @conditional-compile-remove(dateTimeCustomization) */
+    onDisplayDateTimeString
   } = props;
 
+  // onDisplayDateTimeString from messagethread overwrites onDisplayDateTimeString from locale overwrites default
+  let formattedTimestamp: string | undefined = undefined;
+  if (message.createdOn) {
+    // default
+    formattedTimestamp = showDate
+      ? formatTimestampForChatMessage(message.createdOn, new Date(), strings)
+      : formatTimeForChatMessage(message.createdOn);
+
+    /* @conditional-compile-remove(dateTimeCustomization) */
+    if (onDisplayDateTimeString) {
+      formattedTimestamp = onDisplayDateTimeString(message.createdOn);
+    } else if (onDisplayDateTimeStringLocale) {
+      /* @conditional-compile-remove(dateTimeCustomization) */
```

Comment:
Is this conditional compilation flag unnecessary here?
the one above the if block should remove the entire conditional block